src/Server/Repositories/ResultsApi.cs

```diff
@@ -152,9 +144,7 @@ private string GenerateGetPendingJobsUri(string agent, int count)
 
         public void Dispose()
         {
-            _httpClient?.Dispose();
-            _httpClient = null;
-            GC.Collect();
+            _httpClientFactory = null;
```

Comment:
If rusty on C# now, but wonder if you are using the dispose pattern, wouldn't this be needed?
 // Protected implementation of Dispose pattern.
    protected virtual void Dispose(bool disposing)

Reply:
That's only for unmanaged resources.

Reply:
As discussed, setting this private member in a singleton is superfluous.